build_test/lib/src/in_memory_reader.dart

```diff
@@ -9,29 +9,39 @@ import 'package:glob/glob.dart';
 
 import 'in_memory_writer.dart';
 
+/// An [AssetReader] that records which assets have been read to [readAssets].
+abstract class RecordingAssetReader implements AssetReader {
+  Iterable<AssetId> get readAssets;
```

Comment:
nit: `assetsRead`? `readAssets` sounds like an action :)

Reply:
Yeah English is tough. It's hard to distinguish read (present tense) vs read (past tense). I'll change it.